src/core/server/saved_objects/service/lib/update_objects_spaces.test.ts

```diff
@@ -566,68 +562,78 @@ describe('#updateObjectsSpaces', () => {
+      );
       mockBulkResults({ error: false }, { error: false }); // results for obj2 and obj3
 
       const result = await updateObjectsSpaces(params);
       expect(result.objects).toEqual([
-        { ...obj1, spaces: [space2] },
-        { ...obj2, spaces: [space2] },
+        { ...obj1, spaces: [ALL_NAMESPACES_STRING] },
+        { ...obj2, spaces: [otherSpace] },
         { ...obj3, spaces: [] },
       ]);
     });
 
     it('when adding and removing spaces', async () => {
-      const space1 = 'space-to-add';
-      const space2 = 'space-to-remove';
-      const space3 = 'other-space';
-      const obj1 = { type: SHAREABLE_OBJ_TYPE, id: 'id-1', spaces: [space1] }; // will not be changed
-      const obj2 = { type: SHAREABLE_OBJ_TYPE, id: 'id-2', spaces: [space3] }; // will be updated to add space1
-      const obj3 = { type: SHAREABLE_OBJ_TYPE, id: 'id-3', spaces: [space1, space2] }; // will be updated to remove space2
-      const obj4 = { type: SHAREABLE_OBJ_TYPE, id: 'id-4', spaces: [space2, space3] }; // will be updated to add space1 and remove space2
+      const otherSpace = 'space-to-add';
+      const obj1 = { type: SHAREABLE_OBJ_TYPE, id: 'id-1' };
+      const obj2 = { type: SHAREABLE_OBJ_TYPE, id: 'id-2' };
+      const obj3 = { type: SHAREABLE_OBJ_TYPE, id: 'id-3' };
+      const obj4 = { type: SHAREABLE_OBJ_TYPE, id: 'id-4' };
 
       const objects = [obj1, obj2, obj3, obj4];
-      const spacesToAdd = [space1];
-      const spacesToRemove = [space2];
+      const spacesToAdd = [otherSpace];
+      const spacesToRemove = [EXISTING_SPACE];
       const params = setup({ objects, spacesToAdd, spacesToRemove });
-      // this test case does not call mget
+      mockMgetResults(
+        { found: true, namespaces: [ALL_NAMESPACES_STRING, otherSpace] }, // result for obj1 -- will not be changed
+        { found: true, namespaces: [ALL_NAMESPACES_STRING] }, // result for obj2 -- will be updated to add otherSpace
+        { found: true, namespaces: [EXISTING_SPACE, otherSpace] }, // result for obj3 -- will be updated to remove EXISTING_SPACE
+        { found: true, namespaces: [EXISTING_SPACE] } // result for obj4 -- will be updated to remove EXISTING_SPACE and add otherSpace
+      );
       mockBulkResults({ error: false }, { error: false }, { error: false }); // results for obj2, obj3, and obj4
 
       const result = await updateObjectsSpaces(params);
       expect(result.objects).toEqual([
-        { ...obj1, spaces: [space1] },
-        { ...obj2, spaces: [space3, space1] },
-        { ...obj3, spaces: [space1] },
-        { ...obj4, spaces: [space3, space1] },
+        { ...obj1, spaces: [ALL_NAMESPACES_STRING, otherSpace] },
+        { ...obj2, spaces: [ALL_NAMESPACES_STRING, otherSpace] },
+        { ...obj3, spaces: [otherSpace] },
+        { ...obj4, spaces: [otherSpace] },
       ]);
     });
   });
+
+  test.todo('with security extension');
```

Comment:
The security extension actually filters the object graph in addition to redacting space information. See the [old Secure SOC wrapper unit tests](https://github.com/elastic/kibana/blob/8e392a6eed4de5ceef2d0a78e4a10e83ebdce2bc/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.test.ts#L1456-L1475) for details. You should be able to largely copy/paste those tests here.